### PR TITLE
comment on getAttachedObject usage

### PR DESCRIPTION
**Pull request by pavelvasev**

### CODE REVIEW

File: src/qtcore/QMLEngine.js [hunk 1]
```diff
@@ -596,9 +596,31 @@ QMLEngine = function (element, options) {
 
     // TODO: Move to module initialization
     for (i in constructors) {
+
+            /* What's going on here.
+             
+               Some object treated as Attached. For example, Component.
+               In the cycle, we go via constructors and find such objects.
+               Then, we set property to object `QMLBaseObject.prototype` with name of that object, and with specific getter func.
```
Comment (akreuzkamp) [inline src/qtcore/QMLEngine.js]: Please add a line break here :)

File: src/qtcore/QMLEngine.js [hunk 2]
```diff
@@ -596,9 +596,31 @@ QMLEngine = function (element, options) {
...
+               E.g., we create "someitem.Component" here.
+
+               Later, if somebody will read that property, the getter will be invoked. 
+
+               Here all getters are set to `getAttachedObject` only, which is actually dedicated for Component attached object.
+               The code of `getAttachedObject` checks whether $Component internal variable exist, and creates it if it absent.
+               Then, `getAttachedObject` adds self "completed" signal to global `engine.completedSignals`.
+               That is how completed handlers gathered into global list. This list then is called by `engine.callCompletedSignals`.
+
+               memo: setupGetter args: target object, property name, getter function
```
Comment (akreuzkamp) [inline src/qtcore/QMLEngine.js]: This is something, where we imho should trust on our IDEs to provide that information. But feel free to ignore me, if you're into it :)

File: src/qtcore/QMLEngine.js [hunk 3]
```diff
@@ -596,9 +596,31 @@ QMLEngine = function (element, options) {
...
+
+               p.s. At the moment, Repeater and Loader manually call `Component.completed` signals on objects they create.
+                    At the same time, those signals are still pushed to `engine.completedSignals` by getAttachedObject.
+            */
+
```
Comment (akreuzkamp) [inline src/qtcore/QMLEngine.js]: We should add a link to http://doc.qt.io/qt-5/qtqml-syntax-objectattributes.html#attached-properties-and-attached-signal-handlers for further information about attached objects.

**Comment by akreuzkamp** [comment]
Thanks for helping with documentation and sorry for all the nitpicks :)

**Comment by pavelvasev** [comment]
@akreuzkamp is it better now? BTW do you know why QML needs attached properties at all?...

**Comment by ChALkeR** [comment]
@akreuzkamp @pavelvasev What's the status of this?

**Comment by pavelvasev** [comment]
@ChALkeR I think it's still actual comment. Isn't it?..

**Comment by ChALkeR** [comment]
@pavelvasev I was asking because there was some unfinished discussion in the diff comments, but now I see that you amended the commit back then to reflect all the suggestions. I will take another look on this hopefully soon.

**Comment by ChALkeR** [comment]
Landed with minor formatting clean-up in 5c56dd98646aef9267cf6256a2113372a80a4afa, thanks!

**Pull request merged**